repository.go

```diff
@@ -8,122 +8,159 @@ import (
 	sqlStatement :=
-		"select cart_id, user_name " +
-			"from public.cart " +
-			"where cart_id = ?; "
+		"SELECT cart.cart_id, cart.user_name, cart_items.item_amount, item.item_id, item.item_name, item.item_value " +
+			"FROM cart_items, cart, item " +
+			"WHERE cart.cart_id = ? AND cart_items.cart_id = cart.cart_id AND item.item_id = cart_items.item_id; "
 	sqlStatement = strings.Replace(sqlStatement, "?", "1", 1)
 	fmt.Println(sqlStatement)
 
+	var item Item
 	var cart Cart
 	row := db.QueryRow(sqlStatement)
-	switch err := row.Scan(&cart.id, &cart.UserName); err {
+	switch err := row.Scan(&cart.id, &cart.UserName, &item.id, &item.name, &item.value); err {
 	case sql.ErrNoRows:
 		fmt.Println("No rows were returned!")
 	case nil:
+		cart.Products = append(cart.Products, item)
 		fmt.Println(cart)
 	default:
 		panic(err)
 	}
 
 }
 
-func getItem() {
+func listDbItems(db *sql.DB) ([]Item, error) {
+
+	var specificError error
+	sqlStatement := fmt.Sprintf(`
+		select item_id, item_name, item_value 
+		from public.item`)
+
+	fmt.Println(sqlStatement)
 
-	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
-		"password=%s dbname=%s sslmode=disable",
-		host, port, user, password, dbname)
-	db, err := sql.Open("postgres", psqlInfo)
+	var item []Item
+	row, err := db.Query(sqlStatement)
 	if err != nil {
-		panic(err)
+		fmt.Println(err)
+		specificError = err
+	} else {
+		for row.Next() {
+			var newItem Item
+			err := row.Scan(&newItem.id, &newItem.name, &newItem.value)
+			if err != nil {
+				fmt.Println(err)
+				specificError = err
+			}
+			item = append(item, newItem)
+		}
 	}
-	defer db.Close()
 
-	err = db.Ping()
+	fmt.Println(item)
+	return item, specificError
+
+}
+
+func createCartDb(createCart Cart, db *sql.DB) (Cart, error) {
+	var specificError error
+	sqlStatement := fmt.Sprintf(`
+	INSERT INTO public.cart(
+		cart_id, user_name)
+		VALUES (%d, '%s');
+		`, createCart.id, createCart.UserName)
+	fmt.Println(sqlStatement)
+
+	var cart Cart
+	result, err := db.Exec(sqlStatement)
+	_, errInsert := result.RowsAffected()
 	if err != nil {
-		panic(err)
+		fmt.Println(err)
+		specificError = err
+	}
+	if errInsert != nil {
+		fmt.Println(errInsert)
+		specificError = errInsert
 	}
-	fmt.Println("Successfully connected!")
+	return cart, specificError
+}
 
-	sqlStatement :=
-		"select item_id, item_name, item_value " +
-			"from public.item " +
-			"where item_id = ?; "
-	sqlStatement = strings.Replace(sqlStatement, "?", "1", 1)
+func findItem(newItem CreateItem, db *sql.DB) error {
+	sqlStatement := fmt.Sprintf(`
+		select item_id, item_name, item_value 
+		from public.item 
+		where item_id = ?;`, newItem.id)
 	fmt.Println(sqlStatement)
 
 	var item Item
 	row := db.QueryRow(sqlStatement)
-	switch err := row.Scan(&item.id, &item.name, &item.value); err {
-	case sql.ErrNoRows:
-		fmt.Println("No rows were returned!")
-	case nil:
-		fmt.Println(item)
-	default:
-		panic(err)
-	}
-}
+	err := row.Scan(&item.id, &item.name, &item.value)
```

Comment:
Maybe this won't affect the `newItem` variable outside of this function, given that you're receiving it as value (I think)